rocAL/include/pipeline/tensor.h

```diff
@@ -137,8 +137,9 @@ class TensorInfo {
         if (_data_type == data_type)
             return;
         _data_type = data_type;
-        _data_size = (_data_size / _data_type_size);
-        _data_size *= data_type_size();
+        _data_type_size = tensor_data_size(_data_type);
```

Comment:
Is this needed? Are we modifying data type at a later point
The TensorInfo itself will update the stride right?

Reply:
We found this bug during the cosmoflow integration. Log1p kernel accepts int16 inputs but returns f32 outputs. We use the input tensors info but modify the data_type using this function. Without calling modify_strides, the strides were wrong